src/dhtproto/client/DhtClient.d

```diff
@@ -748,7 +748,7 @@ public class DhtClient : IClient
 
     ***************************************************************************/
 
-    private struct Put
+    public struct Put
```

Comment:
It's a shame to have to make this change. It shouldn't be possible to create instances of this struct except via the associated method. I suppose there's no way to enforce this now in a D1/D2 compatible way.

Reply:
Yes, you can eventually mark struct constructor as `@disable this()` in D2. There are still quite some ways to create wrong instance (i.e. `T.init`) but that will cover "accidental" ones.

Here I had to mark it as public because you actually do use this type from other module.

Reply:
Ahh something internal in dhtproto, right. (Do you recall which module that was btw?)

Reply:
No, but it was something defining aliases with those types as template arguments